src/ng/filter/filters.js

```diff
@@ -68,15 +68,36 @@ function currencyFilter($locale) {
       fractionSize = formats.PATTERNS[1].maxFrac;
     }
 
-    // If the currency symbol is empty, trim whitespace around the symbol
-    var currencySymbolRe = !currencySymbol ? /\s*\u00A4\s*/g : /\u00A4/g;
-
     // if null or undefined pass it through
     return (amount == null)
         ? amount
-        : formatNumber(amount, formats.PATTERNS[1], formats.GROUP_SEP, formats.DECIMAL_SEP, fractionSize).
-            replace(currencySymbolRe, currencySymbol);
+        : currencySymbolReReplace(
+            formatNumber(amount, formats.PATTERNS[1], formats.GROUP_SEP, formats.DECIMAL_SEP, fractionSize),
+            currencySymbol
+          );
   };
+
+  function currencySymbolReReplace(input, currencySymbol) {
+    var s = input;
+    if (!input.length) {
+      return s;
+    }
+    // avoid the REDoS by doing this manually, because JS doesn't have positive quantifiers
+    // If the currency symbol is empty, trim whitespace around the symbol
+    var sp = input.split('\u00A4');
+    if (!currencySymbol) {
+      var i;
+      for (i=0; i < sp.length; i++) {
+        if (i > 0) {
+          sp[i] = sp[i].replace(/^\s+/, '');
```

Comment:
pretty minor but might be a little more readable if we used `trimStart` and `trimEnd` instead of regex here and below

Reply:
I chose not to use trimStart and trimEnd because they're not supported in IE, and I wanted to keep to the spirit of Angular.JS's support matrix of IE9 and up.  It's a good instinct for you to check that!